flixel/graphics/frames/bmfontutils/FlxBMFontParser.hx

```diff
@@ -0,0 +1,741 @@
+			var value = kvPair[1];
+			switch key
+			{
+				case 'lineHeight':
+					common.lineHeight = Std.parseInt(value);
+				case 'base':
+					common.base = Std.parseInt(value);
+				case 'scaleW':
+					common.scaleW = Std.parseInt(value);
+				case 'scaleH':
+					common.scaleH = Std.parseInt(value);
+				case 'pages':
+					common.pages = Std.parseInt(value);
+				case 'packed':
+					common.isPacked = value != '0';
+				case 'alphaChnl':
+					common.alphaChnl = Std.parseInt(value);
+				case 'redChnl':
+					common.redChnl = Std.parseInt(value);
+				case 'greenChnl':
+					common.greenChnl = Std.parseInt(value);
+				case 'blueChnl':
+					common.blueChnl = Std.parseInt(value);
+			}
+		}
+		
+		return common;
+	}
+	
+	function parsePageBlock(attrs:String)
+	{
+		var page:BMFontPageInfoBlock = BMFont.getDefaultPageBlock();
+		var i = 0;
+		var word = '';
+		var readNumberLike = () ->
+		{
+			i += 2; // skip '='
+			word = '';
+			while (attrs.charAt(i) != ' ')
+			{
+				word += attrs.charAt(i);
+				i++;
+			}
+			i++; // skip space
+			return word;
+		};
+		var readString = () ->
+		{
+			i += 3; // skip '=' and start quote
+			word = '';
+			while (attrs.charAt(i) != '\"')
+			{
+				word += attrs.charAt(i);
+				i++;
+			}
+			i++; // skip end quote
+			return word;
+		};
+		while (true)
+		{
+			if (i >= attrs.length)
+				break;
+			var curChar = attrs.charAt(i);
+			if (curChar == '')
+			{
+				break;
+			}
+			else if (curChar == ' ')
+			{
+				i++;
+				continue;
+			}
+			
+			word += curChar;
+			switch word
+			{
+				case 'id':
+					page.id = Std.parseInt(readNumberLike());
+					word = '';
+				case 'file':
+					page.file = readString();
+					word = '';
+			}
+			i++;
+		}
+		return page;
+	}
+	
+	function parseCharBlock(attrs:String)
+	{
+		var char:BMFontCharBlock = BMFont.getDefaultCharBlock();
+		var i = 0;
+		var word = '';
+		var readNumberLike = () ->
+		{
+			i += 2; // skip '='
+			word = '';
+			while (attrs.charAt(i) != ' ')
+			{
+				word += attrs.charAt(i);
+				i++;
+			}
+			i++; // skip space
+			return word;
+		};
+		var readString = () ->
+		{
+			i += 3; // skip '=' and start quote
+			// hack needed here specifically because bmfont does not escape double-quotes :(
+			if (attrs.charAt(i) == '"' && attrs.charAt(i + 1) == '"')
+			{
+				return '"';
+			}
+			word = '';
+			while (attrs.charAt(i) != '\"' && i < attrs.length)
+			{
+				word += attrs.charAt(i);
+				i++;
+			}
+			i++; // skip end quote
+			return word;
+		};
+		
+		while (true)
+		{
+			if (i >= attrs.length)
+				break;
+			var curChar = attrs.charAt(i);
+			if (curChar == '')
+				break;
+			if (curChar == ' ')
+			{
+				i++;
+				continue;
+			}
+			word += curChar;
+			var nextChar = attrs.charAt(i + 1);
+			switch word
+			{
+				case 'letter':
+					char.id = Util.getCorrectLetter(readString()).charCodeAt(0);
+					word = '';
+				case 'id':
+					char.id = Std.parseInt(readNumberLike());
+					word = '';
```

Comment:
can you shed some light on this? why set id twice? why give priority to letter?

Reply:
Some files (specially in the text format of bmfont) have a field called `id` which represents the character code of the specific character, while some files use `letter` which contains the character itself. So this should handle both cases